[DO NOT MERGE] Adds page apis migration guide (#9873)
flutter/website: src/release/breaking-changes/navigator-and-page-api.md

---
title: Navigator's page APIs breaking change
description: >
  Changes to Navigator's page APIs.
---

## Summary

The [`Navigator`]'s page APIs are refactored so that they can integrate
into Flutter's other pop mechanisms.

## Context

The `onPopPage` was added for cleaning up pages after a page is about to be popped.

> The developer can veto pop by returning false in the callback. This does not work well

Reviewer: **suggestion:** Condition, then instruction per [GSG](https://developers.google.com/style/sentence-structure). 
Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
To veto pop, the developer can return false in the callback. This does not work well
```

> with other popping mechanisms in the framework, such as [`PopScope`] and cupertino back

Reviewer: **question:** Is the cupertino mention an object or something else? Might be good to link to something.

Reviewer 2: I'd change this even further:

Use the `onPopPage` method to clean up after a page is about to be popped.
You can veto the pop by returning `false` in the callback. This method doesn't work well
with other popping mechanisms in the framework,
such as [`PopScope`][] or [`CupertinoNavigationBarBackButton`][].

[`CupertinoNavigationBarBackButton`]: {{site.api}}/flutter/cupertino/CupertinoNavigationBarBackButton-class.html

gestures.

> The page APIs needed to be refactored in order to integrate them together.

Reviewer: **suggstion:** Condition, then instruction per [GSG](https://developers.google.com/style/sentence-structure).
```suggestion
To integrate the other pop mechanisms together, the page APIs need to be refactored.
```

## Description of change

> The `onPopPage` is replaced by `onDidRemovePage` and no longer had the ability to veto
> a pop. In the `onDidRemovePage`, one is only responsible for updating the [`pages`].

Reviewer: **suggestion:** Write in active voice per [GSG](https://developers.google.com/style/voice). Google Style doesn't recommend using a property as a noun on its own. It should modify what it's changing.

```suggestion
The `onDidRemovePage` property replaces the `onPopPage` property.
It can no longer veto a pop.
In the `onDidRemovePage`, one is only responsible for updating the [`pages`].
```

> The veto mechanism is moved to the `Page.canPop` and `Page.onPopInvoked` similar to how
> one uses the `PopScope` widget.

Reviewer: **suggestion:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
The veto mechanism moves to the `Page.canPop` and `Page.onPopInvoked`. 
These function similar to how one uses the `PopScope` widget.
```
